### PR TITLE
Support legacy _links LFS batch responses

### LINKED ISSUE #31512: LFS mirroring support for legacy batch _links responses
### Feature Description

There are LFS server implementations in the wild which have batch API responses in an older/deprecated schema. Recent versions of Jfrog Artifactory, as well as Microsoft TFS cannot have their LFS objects mirrored into Gitea because they serve a batch API and the 'download' action in responses is not honored because it's under the field `_links` instead of `actions`.

The most recent version of git-lfs still supports this 'deprecated' field: https://github.com/git-lfs/git-lfs/blob/main/tq/transfer.go#L56 , so a `git lfs pull` would work from one of the servers above using the native client, but not with Gitea.

I believe Gitea should support both fields to keep parity with native git-lfs clients until such time as they deprecate it (forcing servers like Artifactory to update their responses).

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by rremer**
Support legacy _links LFS batch response.

Fixes #31512.

This is backwards-compatible change to the LFS client so that, upon mirroring from an upstream which has a batch api, it can download objects whether the responses contain the `_links` field or its successor the `actions` field. When Gitea must fallback to the legacy `_links` field a logline is emitted at INFO level which looks like this:
```
...s/lfs/http_client.go:188:performOperation() [I] <LFSPointer ee95d0a27ccdfc7c12516d4f80dcf144a5eaf10d0461d282a7206390635cdbee:160> is using a deprecated batch schema response!
```

I've only run `test-backend` with this code, but added a new test to cover this case. Additionally I have a fork with this change deployed which I've confirmed syncs LFS from Gitea<-Artifactory (which has legacy `_links`) as well as from Gitea<-Gitea (which has the modern `actions`).

### CODE REVIEW

File: modules/lfs/http_client.go [hunk 1]
```diff
@@ -179,8 +179,17 @@ func (c *HTTPClient) performOperation(ctx context.Context, objects []Pointer, dc
...
 			}
 		} else {
+			var link *Link
 			link, ok := object.Actions["download"]
 			if !ok {
+				// no actions block in response, try legacy response schema
+				link, ok = object.Links["download"]
+				if ok {
+					log.Info("%+v is using a deprecated batch schema response!", object)
+				}
```
Comment (wxiaoguang) [inline modules/lfs/http_client.go]: Well, this "Info" log should be removed 😄 

```suggestion
```

File: modules/lfs/http_client.go [hunk 2]
```diff
@@ -179,8 +179,17 @@ func (c *HTTPClient) performOperation(ctx context.Context, objects []Pointer, dc
 				}
 			}
 		} else {
+			var link *Link
```
Comment (wxiaoguang) [inline modules/lfs/http_client.go]: According to Golang syntax, this line could be removed.

```suggestion
```

**Comment by rremer** [comment]
thanks for the review @wxiaoguang , updated with your suggestions and force pushed

**Review by wxiaoguang — APPROVED** [review]

File: modules/lfs/http_client.go [hunk 3]
```diff
@@ -180,6 +180,10 @@ func (c *HTTPClient) performOperation(ctx context.Context, objects []Pointer, dc
 			}
 		} else {
 			link, ok := object.Actions["download"]
+			if !ok {
+				// no actions block in response, try legacy response schema
+				link, ok = object.Links["download"]
```
Comment (lunny) [inline modules/lfs/http_client.go:185]: Should we check whether `Links` is nil?

**Review by lunny — APPROVED** [review]

**Pull request merged**